lib/modules/manager/glasskube/extract.ts

```diff
@@ -0,0 +1,119 @@
+import { readLocalFile } from '../../../util/fs';
+import { parseYaml } from '../../../util/yaml';
+import { GlasskubePackagesDatasource } from '../../datasource/glasskube-packages';
+import type {
+  ExtractConfig,
+  PackageDependency,
+  PackageFile,
+  PackageFileContent,
+} from '../types';
+import { GlasskubeResource, Package, PackageRepository } from './schema';
+import type { GlasskubeResources } from './types';
+
+function parseResources(
+  content: string,
+  packageFile: string,
+): GlasskubeResources {
+  const resources: GlasskubeResource[] = parseYaml(content, null, {
+    json: true,
+    customSchema: GlasskubeResource,
+    failureBehaviour: 'filter',
+  });
+
+  const packages: Package[] = [];
+  const repositories: PackageRepository[] = [];
+
+  for (const resource of resources) {
+    if (resource.kind === 'ClusterPackage' || resource.kind === 'Package') {
+      packages.push(resource);
+    } else if (resource.kind === 'PackageRepository') {
+      repositories.push(resource);
+    }
+  }
+
+  return { packageFile, repositories, packages };
+}
+
+function resolvePackageDependencies(
+  packages: Package[],
+  repositories: PackageRepository[],
+): PackageDependency[] {
+  const deps: PackageDependency[] = [];
+  for (const pkg of packages) {
+    const dep: PackageDependency = {
+      depName: pkg.spec.packageInfo.name,
+      currentValue: pkg.spec.packageInfo.version,
+      datasource: GlasskubePackagesDatasource.id,
+    };
+
+    const repository = findRepository(
+      pkg.spec.packageInfo.repositoryName ?? null,
+      repositories,
+    );
+
+    if (repository === null) {
+      dep.skipReason = 'unknown-registry';
+    } else {
+      dep.registryUrls = [repository.spec.url];
+    }
+
+    deps.push(dep);
+  }
+  return deps;
+}
+
+function findRepository(
+  name: string | null,
+  repositories: PackageRepository[],
+): PackageRepository | null {
+  return (
+    repositories.find(
+      (it) =>
+        name === it.metadata.name ||
+        ((name === null || name === '') && isDefaultRepository(it)),
+    ) ?? null
+  );
+}
+
+function isDefaultRepository(repository: PackageRepository): boolean {
+  return (
+    repository.metadata.annotations?.[
+      'packages.glasskube.dev/default-repository'
+    ] === 'true'
+  );
+}
+
+export function extractPackageFile(
+  content: string,
+  packageFile: string,
+  config?: ExtractConfig,
+): PackageFileContent | null {
+  const { packages, repositories } = parseResources(content, packageFile);
+  const deps = resolvePackageDependencies(packages, repositories);
+  return { deps };
+}
```

Comment:
```suggestion
```
This will never be called as you have defined `extractAllPackageFiles`

Reply:
it's needed for auto replace

Reply:
I don't know exactly why it's needed but I took the info from https://github.com/renovatebot/renovate/blob/main/docs/development/adding-a-package-manager.md, where it states this:

> If you implement `extractAllPackageFiles` the manager must export as well either `updateDependency` or `extractPackageFile`.